packages/trampoline/src/trampoline.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable @jessie.js/safe-await-separator */
+/**
+ * @import {AnyGenerator, SyncTrampolineGeneratorFn, AsyncTrampolineGeneratorFn } from './types.js'
+ */
+
+/**
+ * Trampoline on {@link TrampolineGeneratorFn generatorFn} synchronously.
+ *
+ * @template {readonly any[]} [TArgs=unknown[]] Parameters for `generatorFn`
+ * @template [TResult=unknown] Type of the return value of the `generatorFn`
+ * @template {Generator<any, any, TResult>} [TGenerator=Generator] Type of the generator function
+ * @param {SyncTrampolineGeneratorFn<TGenerator, TArgs, TResult>} generatorFn Generator-returning function accepting a thunk and optionally an initial value
+ * @param {TArgs} args Initial args
+ * @returns {TResult}
+ */
+export function syncTrampoline(generatorFn, ...args) {
+  const iterator = generatorFn(...args);
+  let result = iterator.next();
+  while (!result.done) {
+    result = iterator.next(result.value);
+  }
+  return result.value;
+}
+
+/**
+ * Trampoline on {@link TrampolineGeneratorFn generatorFn} asynchronously.
+ *
+ * @template {readonly any[]} [TArgs=unknown[]] Parameters for `generatorFn`
+ * @template [TResult=unknown] Type of the return value of the `generatorFn`
+ * @template {Generator<any, any, TResult>} [TGenerator=Generator] Type of the generator function
+ * @param {AsyncTrampolineGeneratorFn<TGenerator, TArgs, TResult>} generatorFn Generator-returning function accepting a thunk and optionally an initial value
+ * @param {TArgs} args Initial args
+ * @returns {Promise<TResult>}
+ */
+export async function trampoline(generatorFn, ...args) {
+  const iterator = generatorFn(...args);
+  let result = iterator.next();
```

Comment:
If we want to use this in SES we'll need to uncuryThis on the next and error methods and keep them local. (can steal the 3 lines for that from commons in SES)

Reply:
@naugtur I don't understand if this is actionable by me in this PR?